src/Orckestra.Composer.CompositeC1/Context/ProductTileConfigurationContext.cs

```diff
@@ -0,0 +1,76 @@
+﻿using Composite.Data;
+using Orckestra.Composer.CompositeC1.DataTypes;
+using Orckestra.Composer.CompositeC1.Services.Cache;
+using Orckestra.Composer.CompositeC1.Services.DataQuery;
+using System.Collections.Generic;
+using System.Web;
+
+namespace Orckestra.Composer.CompositeC1.Context
+{
+    public class ProductTileConfigurationContext : IProductTileConfigurationContext
+    {
+        protected ICacheStore<string, List<IPromotionalRibbonConfiguration>> PromotionalRibbonCache { get; }
+        protected ICacheStore<string, List<IPromotionalBannerConfiguration>> PromotionalBannerCache { get; }
+        protected HttpContextBase HttpContext { get; }
+        protected IDataQueryService DataQueryService { get; }
+        private List<IPromotionalRibbonConfiguration> _promotionalRibbonConfigurations;
+        public string PromotionalRibbonDefaultBackgroundColor => "bg-dark";
+        public string PromotionalRibbonDefaultTextColor => "text-white";
+        private List<IPromotionalBannerConfiguration> _promotionalBannerConfigurations;
+        public string PromotionalBannerDefaultBackgroundColor => "bg-dark";
+        public string PromotionalBannerDefaultTextColor => "text-white";
+
+        public ProductTileConfigurationContext(HttpContextBase httpContext, IDataQueryService dataQueryService, ICacheService cacheService)
+        {
+            HttpContext = httpContext;
+            DataQueryService = dataQueryService;
+            PromotionalRibbonCache = cacheService.GetStoreWithDependencies<string, List<IPromotionalRibbonConfiguration>>("Promotional Ribbons",
+                new CacheDependentEntry<IPromotionalRibbonConfiguration>());
+
+            PromotionalBannerCache = cacheService.GetStoreWithDependencies<string, List<IPromotionalBannerConfiguration>>("Promotional Banners",
+                new CacheDependentEntry<IPromotionalBannerConfiguration>());
+        }
+        public List<IPromotionalRibbonConfiguration> GetPromotionalRibbonConfigurations()
+        {
+            if (_promotionalRibbonConfigurations == null)
```

Comment:
Why do we not return PromotionalRibbonCache.GetOrAdd("PromotionalRibbonSettings", _ => LoadPromotionalRibbonSettings())
The class registered as Transistion and _promotionalRibbonConfigurations will be null everytime